rrdd/design/plugin-protocol-v2.md

```diff
@@ -96,14 +97,27 @@ Protocol V2
 |metadata checksum    |32               |int32 |binary-encoded crc32 of the metadata string (see below)                                 |
 |number of datasources|32               |int32 |only needed if the metadata has changed - otherwise RRDD can use a cached value         |
 |timestamp            |64               |int64 |Unix epoch                                                                              |
-|datasource values    |n * 64           |int64 |n is the number of datasources exported by the plugin                                   |
+|datasource values    |n * 64           |int64 \| double |n is the number of datasources exported by the plugin, type dependent on the setting in the metadata for value_type [int64\|float]  |
 |metadata length      |32               |int32 |                                                                                        |
 |metadata             |(string length)*8|string|                                                                                        |
 
-All integers are bigendian. The metadata will have the same JSON-based format as
+All integers/double are bigendian. The metadata will have the same JSON-based format as
 in the V1 protocol, minus the timestamp and `value` key-value pair for each
-datasource, for example:
+datasource.
 
+| field | values | notes | required |
+|-------|--------|-------|----------|
+|description|string|Description of the datasource|no|
+|owner|host \| vm \| sr|The object to which the data relates|no, default host|
+|value_type|int64 \| float|The type of the datasource|no, default float|
```

Comment:
there is no default AFAICT. xcp-rrdd will internally raise an exception if 'value_type' is missing

Reply:
OK, I was going off the python binding :) which does default to float.